app/src/main/java/br/com/dionataferraz/vendas/data/remote/RetrofitNetworkClient.kt

```diff
@@ -0,0 +1,55 @@
+package br.com.dionataferraz.vendas.data.remote
+
+import br.com.dionataferraz.vendas.BuildConfig
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.util.concurrent.TimeUnit
+
+const val DURATION = 60L
+
+object RetrofitNetworkClient {
+
+    fun createNetworkClient(baseUrl: String = BuildConfig.HTTP_SERVER): Retrofit =
+        retrofitClient(
+            baseUrl = baseUrl,
+            httpClient = httpClient(),
+            moshiConverter = moshi(),
+        )
+
+    private fun moshi() = MoshiConverterFactory.create(
+        Moshi.Builder()
+            .add(KotlinJsonAdapterFactory())
+            .build()
+    )
```

Comment:
Para funcionar a listagem,faltou essa dependencia
```suggestion
    private fun moshi() = MoshiConverterFactory.create(
        Moshi.Builder()
            .add(KotlinJsonAdapterFactory())
            .add(Date::class.java, Rfc3339DateJsonAdapter().nullSafe())
            .build()
    )
```